src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs

```diff
@@ -69,6 +69,12 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
 
             var arg1 = argTypes[1];
 
+            // coerce number to decimal and vice versa.
```

Comment:
>// coerce number to decimal and vice versa. [](http://example.com/codeflow?start=12&length=43)

Why limit coercion to numbers only? Can we allow coercions to other types (e.g., set a GUID to a string variable)?

Reply:
Higher chance of storing runtime errors unknowingly for other types, so I thought initially we could start with numeric and that behavior change feels _almost_ transparent to the end user. 
I am open to expanding it to other types. What's your advice?

Reply:
I'm leaning towards just numbers, but on the fence. 

 

I know we originally hoped dec/float was "just" ordinary coercion. But we really do have 2 concepts here:

Numeric coercion (float-->dec) is "closer" than general coercion (str-->float). 

 

general coercion is really more of "pasring" than just coercion. And practically, it has a higher chance of maker error, requires a locale, and can introduce more runtime errors that could have been caught at design time.